### ISSUE TITLE
improve support for nested JSON structures

### ISSUE THREAD

**Issue by tzemanovic** [mrkkrp/forma#8]
Hi Mark,

Thanks for this library! I'm trying to use it for validation on a nested JSON, e.g.:

```json
{
  "user":{
    "email": "fanny@foo.com",
    "password": "secret"
  }
}
```

I think I'd need to have FormParser for each level:

```haskell
userForm :: Monad m => FormParser UserFields m Value
userForm =
  field' @"user"

loginForm :: Monad m => FormParser LoginFields m Login
loginForm = Login
    <$> field @"email" notEmpty
    <*> field @"password" notEmpty
```

Unless I'm missing something, this is a bit awkward, because then I'd have to `runForm` for each `FormParser` and combine the results.

**Comment by mrkkrp** [comment]
Indeed, that's currently not supported. It's not hard to add support for this, but then we'll need to release new major version because this change, while looking trivial at the surface, necessarily makes field names multi-component objects instead of flat `Text` values. I.e. I can have a validation error not just for `"email"` field, but for a field that is nested inside other field, and must be different than a field with the same name but different parent (`"user.email"` vs `"foo.email"`).

**Comment by mrkkrp** [comment]
Please see if #9 solves the problem.

**Comment by tzemanovic** [comment]
Thanks, that does it. Please see #10 before merge.
